s3po/backends/s3.py

```diff
@@ -107,3 +107,17 @@ def list(self, bucket, prefix=None, delimiter=None, retries=3, headers=None):
         # consume iterator to make a list to keep parity with Swift backend
         return (key.name for key in self._list_retry(retries, bucket,
                                           prefix, delimiter, headers=headers))
+
+    def delete(self, bucket, key, retries, headers=None):
+        '''Delete bucket/key with headers'''
+        # Make our headers object
+        headers = headers or {}
+        bucket = self.get_bucket(bucket)
+        key = bucket.new_key(key)
+
+        @retry(retries)
+        def func():
+            '''The bit that we want to retry'''
+            key.delete(headers=headers)
```

Comment:
It is not clear from [documentation](http://boto.cloudhackers.com/en/latest/ref/s3.html#module-boto.s3.key) that `key.delete()` might raise exception. I read it several times.